src/postgres.c

```diff
@@ -27,14 +29,19 @@ static struct db_operations pg_db_ops = {
 int construct_pg(void)
 {
 	struct db_t *pg_db_t = CNC_MALLOC(sizeof(struct db_t));
-	email_body = CNC_MALLOC(EMAIL_BODY_LENGTH * sizeof(char));
+
+	if (log_file == NULL) {
+		free(pg_db_t);
+		return -1;
+	}
```

Comment:
This seems wrong. We shouldn't have reached this place if the `log_file` is **NULL**.

Reply:
That is true but there is a case that we can reach this and we get segfault.
If the user used `sudo` rights to create a folder that need permissions, but then used tried to
create a log file to that folder without `sudo` writes then we cant actually open the file, therefore it is `NULL`.
This is not a common case but we can 100% sure that we don't get a segfault even if the user used an uncommon
pattern for creating the folder and log files.